src/pages/Auth/SignUp.jsx

```diff
@@ -0,0 +1,85 @@
+import { MdEmail } from "react-icons/md";
+import { RiLockPasswordFill } from "react-icons/ri";
+import { FaUserCircle } from "react-icons/fa";
+import { AiFillEye, AiFillEyeInvisible } from "react-icons/ai";
+import { Link, useNavigate } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { useAuth } from "../../contexts/auth-context";
+import { ACTION_TYPE } from "../../utils";
+import { Header } from "../../components";
+
+export const SignUp = () => {
+
+    const [signupFormData, setsignupFormData] = useState({
+        userName: "",
+        email: "",
+        password: "",
+        passType: "password",
+    })
+
+    const { authState: { authToken, error }, signUpFormHandler, authDispatch } = useAuth();
+    const { SET_ERROR } = ACTION_TYPE;
+    const navigate = useNavigate();
+
+    const formSubmitHandler = (event) => {
+        event.preventDefault();
+        signUpFormHandler(signupFormData);
+    };
+
+    // password visibilty handler here 
+    const passVisibiltyHandler = () => {
+        return signupFormData.passType === "password" ? setsignupFormData({ ...signupFormData, passType: "text" }) : setsignupFormData({ ...signupFormData, passType: "password" });
+    }
+
+    useEffect(() => {
+        if (authToken) { navigate('/products') }
```

Comment:
here also you can provide dynamic page navigate

Reply:
Yeah sure.